### DISCUSSION TITLE
Lanbon L8 HD Serial Protocol

### DISCUSSION THREAD

**Discussion by Beormund** [HASwitchPlate/openHASP#380] [Q&A] [answered]
Hi,  I'm writing some Tasmota Berry code to control dimming of the Lanbon L8. Would anyone be able to provide the serial hex codes  / protocol please? I'm looking for on, off, and % dimming. Much appreciated.

**Comment by fvanroie** [comment, marked as answer]
Check out this HA forum post: https://community.home-assistant.io/t/interesting-touch-switch-lanbon-l8/264112/40

The 4-byte serial communication is using TX => K3 – IO12 (115200 baud, 8data, 1stop, no parity).

**Reply by Beormund** [reply]
So to initialise dimmer: `"\0x20\0x20\\0xEF\0x01\0x4D\0xA3"`
Set to OFF: `"\0xEF\0x02\0x20\0xED"`

Set to 70%: `"\0xEF\0x02\0x46\0xAB"` (Byte 4: 0x46 ^ 0xED = 0xAB) 
Set to 22%: `"\0xEF\0x02\0x16\0xFB"` (Byte 4:  0x16 ^ 0xED = 0xFB)
